solidity/contracts/utility/MocBTCToUSDOracle.sol

```diff
@@ -36,10 +37,12 @@ contract MocBTCToUSDOracle is IConsumerPriceOracle, Owned {
     /**
       * @dev returns the USD/BTC update time.
       *
-      * @return always returns current block's timestamp
+      * @return returns the latest block's timestamp
     */
     function latestTimestamp() external view returns (uint256) {
-        return now; // MoC oracle doesn't return update timestamp
+        require(block.number >= Medianizer(mocOracleAddress).getLastPublicationBlock(), "latest block number larger than current block numer");
+        uint256 latestTimestamp_ = block.timestamp - ( block.number - Medianizer(mocOracleAddress).getLastPublicationBlock() ) * 30;
+        return latestTimestamp_; 
```

Comment:
You're calling Medianizer(mocOracleAddress).getLastPublicationBlock() twice. It would be better to read it only once and store it in a local variable to save gas.